detection-rules/impersonation_twitter.yml

```
@@ -1,18 +1,42 @@
 name: "Brand impersonation: Twitter"
 description: |
-  Impersonation of Twitter
+  Impersonation of Twitter.
 references:
   - "https://www.techrepublic.com/article/phishing-attack-spoofs-twitter-to-steal-account-credentials/"
 type: "rule"
 severity: "medium"
 source: |
   type.inbound
   and (
-    sender.display_name =~ "twitter"
-    or strings.ilevenshtein(sender.display_name, 'twitter') <= 1
-    or strings.ilike(sender.email.domain.domain, '*twitter*')
+    // Twitter logic
+    (
+      sender.display_name =~ "twitter"
+      or strings.ilevenshtein(sender.display_name, 'twitter') <= 1
+      or strings.ilike(sender.email.domain.domain, '*twitter*')
+    )
+    // "X" logic
+    or (
+      any(ml.logo_detect(beta.message_screenshot()).brands,
+          .name == "X" and .confidence == "high"
+      )
+      and any(ml.nlu_classifier(body.current_thread.text).intents,
+              .name == "cred_theft" and .confidence == "high"
+      )
+      and any(beta.ml_topic(body.current_thread.text).topics,
+                  .name in (
+                    "Reminders and Notifications",
```

might tack on `"Security and Authentication"`
```suggestion
                    "Reminders and Notifications",
                    "Security and Authentication",
```